http/examples/http_middleware.rs

```diff
@@ -0,0 +1,29 @@
+extern crate jsonrpc_core;
+extern crate jsonrpc_http_server;
+
+
+use jsonrpc_core::*;
+use jsonrpc_core::futures::Future;
+use jsonrpc_http_server::*;
+
+fn main() {
+	let mut io = IoHandler::default();
+	io.add_async_method("say_hello", |_params| {
+		futures::finished(Value::String("hello".to_owned())).boxed()
+	});
+
+	let server = ServerBuilder::new(io)
+		.cors(DomainsValidation::AllowOnly(vec![AccessControlAllowOrigin::Null]))
+		.request_middleware(|request: &hyper::server::Request<hyper::net::HttpStream>| {
```

Comment:
That was the thing where the compiler got confused about bound lifetimes without explicit type in the closure?